src/3d/framegraph/qgsframegraph.cpp

```diff
@@ -81,85 +83,24 @@ void QgsFrameGraph::constructDebugTexturePass( Qt3DRender::QFrameGraphNode *topN
   registerRenderView( std::make_unique<QgsDebugTextureRenderView>( DEBUG_RENDERVIEW ), DEBUG_RENDERVIEW, topNode );
 }
 
-Qt3DRender::QFrameGraphNode *QgsFrameGraph::constructSubPostPassForProcessing()
+void QgsFrameGraph::constructPostprocessingPass()
 {
-  Qt3DRender::QCameraSelector *cameraSelector = new Qt3DRender::QCameraSelector;
-  cameraSelector->setObjectName( "Sub pass Postprocessing" );
-  cameraSelector->setCamera( shadowRenderView().lightCamera() );
+  // create post processing render view and register it
+  QgsPostprocessingRenderView *pprv = new QgsPostprocessingRenderView( POSTPROC_RENDERVIEW,          //
+                                                                       shadowRenderView(),           //
+                                                                       forwardRenderView(),          //
+                                                                       ambientOcclusionRenderView(), //
+                                                                       mSize,                        //
+                                                                       mRootEntity );
+  registerRenderView( std::unique_ptr<QgsPostprocessingRenderView>( pprv ), POSTPROC_RENDERVIEW );
 
-  Qt3DRender::QLayerFilter *layerFilter = new Qt3DRender::QLayerFilter( cameraSelector );
+  // create debug texture render view and register it (will be detach from main framegraph and attach to postprocessing renderview)
+  constructDebugTexturePass();
```

Comment:
why would debug texture pass need to be attached to post-processing render view? it can be a render view by itself